### PR TITLE
[ Application ] Make Utils for Application @open sesame 09/23 10:21

### PR DESCRIPTION
**Pull request by jijoongmoon**
Currently some of application use bitmap_helper and they have it
independently. rather than do this way, It might be better to make
static lib and link.

**Self evaluation:**
1. Build test:	 [X]Passed [ ]Failed [ ]Skipped
2. Run test:	 [X]Passed [ ]Failed [ ]Skipped

Signed-off-by: jijoong.moon <jijoong.moon@samsung.com>

### CODE REVIEW

File: Applications/utils/jni/meson.build [hunk 1]
```diff
@@ -0,0 +1,24 @@
+utils_inc = include_directories('./includes')
```
Comment (kparichay) [inline Applications/utils/jni/meson.build:1]: Android.mk is missing for this? I mean how does Android.mk of application gets the libnntrainer-util.so from?

Comment (jijoongmoon) [reply Applications/utils/jni/meson.build:1]: fixed. thanks.

File: Applications/utils/jni/meson.build [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+]
+
+shared_library('nntrainer_utils',
+	nntrainer_utils_source,
+	include_directories: utils_inc,
+	install: get_option('install-app'),
+	install_dir: application_install_dir
+)
+
+nntrainer_utils_lib = static_library('nntrainer_utils',
```
Comment (zhoonit) [inline Applications/utils/jni/meson.build:14]: Could we rename this as nntrainer_app_utils or equivalent?
nntrainer_utils sounds like it is part of the library (if it is planned to be, I think it's fine)

File: Applications/KNN/jni/Android.mk [hunk 3]
```diff
@@ -8,7 +8,7 @@ $(error ANDROID_NDK is not defined!)
 endif
 
 ifndef NNTRAINER_ROOT
-NNTRAINER_ROOT := $(LOCAL_PATH)/../../../libs/arm64-v8a
+NNTRAINER_ROOT := $(LOCAL_PATH)/../../../
 NNTRAINER_INCLUDES := $(LOCAL_PATH)/../../../nntrainer/include \
 	$(LOCAL_PATH)/../../../api \
 	$(LOCAL_PATH)/../../../api/capi/include/platform
```
Comment (zhoonit) [inline Applications/KNN/jni/Android.mk:14]: Minor suggestion :)
```suggestion
NNTRAINER_INCLUDES := $(NNTRAINER_ROOT)/nntrainer/include \
  $(NNTRAINER_ROOT)/api \
	$(NNTRAINER_ROOT)/api/capi/include/platform
```

File: Applications/utils/jni/meson.build (hunk 2 above)
Comment (jijoongmoon) [reply Applications/utils/jni/meson.build]: fixed. thanks

File: Applications/KNN/jni/Android.mk (hunk 3 above)
Comment (jijoongmoon) [reply Applications/KNN/jni/Android.mk]: fixed. thanks

File: Applications/TransferLearning/CIFAR_Classification/jni/Android.mk [hunk 4]
```diff
@@ -48,6 +48,14 @@ include $(PREBUILT_SHARED_LIBRARY)
 
 include $(CLEAR_VARS)
 
+LOCAL_MODULE := nntrainer_utils
```
Comment (kparichay) [inline Applications/TransferLearning/CIFAR_Classification/jni/Android.mk:51]: app_utils?

File: Applications/TransferLearning/CIFAR_Classification/jni/Android.mk [hunk 5]
```diff
@@ -48,6 +48,14 @@ include $(PREBUILT_SHARED_LIBRARY)
...
+LOCAL_SRC_FILES := $(NNTRAINER_ROOT)/Applications/utils/libs/$(TARGET_ARCH_ABI)/libnntrainer_utils.so
```
Comment (kparichay) [inline Applications/TransferLearning/CIFAR_Classification/jni/Android.mk:52]: libapp_utils.so?

Comment (kparichay) [reply Applications/TransferLearning/CIFAR_Classification/jni/Android.mk:52]: I wonder how this even build successfully?

**Review by zhoonit — APPROVED** [review]
LGTM!

File: Applications/TransferLearning/CIFAR_Classification/jni/Android.mk (hunk 5 above)
Comment (jijoongmoon) [reply Applications/TransferLearning/CIFAR_Classification/jni/Android.mk]: I checked the CI and it is fixed.

File: Applications/TransferLearning/CIFAR_Classification/jni/Android.mk (hunk 4 above)
Comment (jijoongmoon) [reply Applications/TransferLearning/CIFAR_Classification/jni/Android.mk]: fixed.

**Review by kparichay — APPROVED** [review]
LGTM

**Pull request merged**